.github/workflows/push_to_main.yml

```diff
@@ -0,0 +1,9 @@
+name: Push to Main CI
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
Here, I feel more strongly that CI should only be run on pushes to the main branch. We were running CI twice for PRs (once on branch push, once for the PR).